### PR TITLE
feat: external argo list

### PR DESCRIPTION
**Pull request by shivani170**
# Description

To add extrenal argo list

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/list-new/ExternalArgoList.tsx [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+    clearAllFilters,
+    fetchingExternalApps,
+    setFetchingExternalAppsState,
+    updateDataSyncing,
+    setShowPulsatingDotState,
+    masterFilters,
+    syncListData,
+    isArgoInstalled,
+}: ExternalArgoListType) {
+    console.log(
```
Comment (vivek-devtron) [inline src/components/app/list-new/ExternalArgoList.tsx:64]: Please remove console.log

File: src/components/app/list-new/AppList.tsx [hunk 2]
```diff
@@ -47,6 +48,8 @@ export default function AppList({ isSuperAdmin, appListCount, isArgoInstalled }:
     const [currentTab, setCurrentTab] = useState(undefined)
     const [syncListData, setSyncListData] = useState<boolean>()
     const [projectMap, setProjectMap] = useState(new Map());
+
+    const isExternalArgo = location.pathname === "/app/list/a"
```
Comment (vivek-devtron) [inline src/components/app/list-new/AppList.tsx:52]: We should use constants for this

File: src/components/app/list-new/ExternalArgoList.tsx [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+    useEffect(() => {
+        if (dataStateType == AppListViewType.LIST) {
+            handleFilteration()
+        }
+    }, [dataStateType])
+
+    useEffect(() => {
+        updateDataSyncing(true)
+        setDataStateType(AppListViewType.LOADING)
+        if (serverMode == SERVER_MODE.EA_ONLY) {
```
Comment (vivek-devtron) [inline src/components/app/list-new/ExternalArgoList.tsx:116]: Is it required?

File: src/components/app/list-new/ExternalArgoList.tsx [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+            setDataStateType(AppListViewType.LIST)
+            if (clusterIdsCsv) {
+                _getExternalHelmApps()
```
Comment (vivek-devtron) [inline src/components/app/list-new/ExternalArgoList.tsx:119]: Same here

File: src/components/app/list-new/ExternalArgoList.tsx [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+    )
+    const [dataStateType, setDataStateType] = useState(AppListViewType.LOADING)
+    const [errorResponseCode, setErrorResponseCode] = useState(0)
+    const [argoAppsList, setArgoAppsList] = useState<ArgoAppListResult[]>([])
+    const [filteredArgoAppsList, setFilteredArgoAppsList] = useState<ArgoAppListResult[]>([])
+    const [sortBy, setSortBy] = useState(SortBy.APP_NAME)
+    const [sortOrder, setSortOrder] = useState(OrderBy.ASC)
+    const [clusterIdsCsv, setClusterIdsCsv] = useState('')
+    const [appStatus, setAppStatus] = useState('')
+    const [showGuidedContentCards, setShowGuidedContentCards] = useState(false)
```
Comment (vivek-devtron) [inline src/components/app/list-new/ExternalArgoList.tsx:85]: is this state required?

File: src/components/app/list-new/ExternalArgoList.tsx [hunk 6]
```diff
@@ -0,0 +1,494 @@
...
+    function renderApplicationList() {
+        return (
+            <div data-testid="helm-app-list-container">
+                {filteredArgoAppsList.length > 0 && renderHeaders()}
+
+                {filteredArgoAppsList
+                    .slice(payloadParsedFromUrl.hOffset, payloadParsedFromUrl.hOffset + payloadParsedFromUrl.size)
+                    .map((app) => renderHelmAppLink(app))}
+
+                {showGuidedContentCards && (
```
Comment (vivek-devtron) [inline src/components/app/list-new/ExternalArgoList.tsx:295]: Is it required here?

File: src/components/app/list-new/ExternalArgoList.tsx [hunk 7]
```diff
@@ -0,0 +1,494 @@
...
+                    subTitle={APPLIST_EMPTY_STATE_MESSAGING.connectClusterInfoText}
+                    isButtonAvailable={true}
+                    renderButton={handleButton}
+                />
+            </div>
+        )
+    }
+
+    function renderHelmPermissionMessageStrip() {
+        return (
```
Comment (vivek-devtron) [inline src/components/app/list-new/ExternalArgoList.tsx:403]: Is it required?

File: src/components/app/list-new/ExternalArgoList.tsx [hunk 8]
```diff
@@ -0,0 +1,494 @@
...
+        } else {
+            return renderAllCheckModal()
+        }
+    }
+
+    function renderFullModeApplicationListContainer() {
+        if (filteredArgoAppsList.length == 0) {
+            return (
+                <>
+                    {serverMode == SERVER_MODE.FULL && renderHelmPermissionMessageStrip()}
```
Comment (vivek-devtron) [inline src/components/app/list-new/ExternalArgoList.tsx:434]: Is this required?

**Comment by shivani170** [comment]
Customer demo Feature only

**Pull request closed without merging**